ADOs/nft-staking/src/state.rs

```diff
@@ -0,0 +1,131 @@
+use andromeda_std::common::Milliseconds;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{ensure, Addr, BlockInfo, Storage, Uint128};
+use cw_storage_plus::{Item, Map};
+
+use crate::ContractError;
+
+#[cw_serde]
+#[derive(Default)]
+pub struct Config {
+    pub denom: String,
+    pub unbonding_period: Milliseconds,
+    pub payout_window: Milliseconds,
+}
+
+#[cw_serde]
+#[derive(Default)]
+pub struct StakerDetail {
+    // list of staked nft assets as (nft_address, nft_id)
+    pub assets: Vec<(String, String)>,
+}
+
+#[cw_serde]
+#[derive(Default)]
+pub struct AssetDetail {
+    pub nft_address: String,
+    pub token_id: String,
+    pub unbonding_period: Milliseconds,
+    pub pending_rewards: Uint128,
+    pub updated_at: Milliseconds,
+    pub unstaked_at: Option<Milliseconds>,
+}
+
+pub const CONFIG: Item<Config> = Item::new("config");
+
+pub const REWARDS_PER_TOKEN: Map<&str, Uint128> = Map::new("rewards_per_token");
+
+pub const STAKER_DETAILS: Map<&Addr, StakerDetail> = Map::new("staker_details");
+
+pub const ASSET_DETAILS: Map<(String, String), AssetDetail> = Map::new("asset_details");
+
+pub fn set_rewards_per_token(
+    store: &mut dyn Storage,
+    rewards_per_token: Vec<(String, u128)>,
+) -> Result<(), ContractError> {
+    rewards_per_token.iter().for_each(|item| {
+        REWARDS_PER_TOKEN
+            .save(store, &item.0, &Uint128::from(item.1))
+            .unwrap();
```

Comment:
Replace instances of unwrap with `?`

Reply:
`Result<(), ContractError>`
Maybe it is right to use upwrap instance for this return types.